placesautocomplete/src/main/java/com/seatgeek/placesautocomplete/network/OkHttpPlacesHttpClient.java

```diff
@@ -32,15 +32,19 @@
 
         Response response = okHttpClient.newCall(request).execute();
 
-        T body = responseHandler.handleStreamResult(response.body().byteStream());
-
-        Status status = body.status;
-
-        if (status != null && !status.isSuccessful()) {
-            String err = body.error_message;
-            throw new PlacesApiException(err != null ? err : "Unknown Places Api Error");
-        } else {
-            return body;
+        try {
+            T body = responseHandler.handleStreamResult(response.body().byteStream());
+            Status status = body.status;
+            if (status != null && !status.isSuccessful()) {
+                String err = body.error_message;
+                throw new PlacesApiException(err != null ? err : "Unknown Places Api Error");
+            } else {
+                return body;
+            }
+        } finally {
+            if (response != null) {
+                response.body().close();
```

Comment:
Does this need to be in a try/catch to handle failures closing?

Reply:
I think it´s best that way since the method also throw new PlacesApiException

Reply:
Like this?:

    try {
            response.body().close();
        }catch(Exception e){
            Log.w(Constants.LOG_TAG, "Exception Closing Response body..", e);
        }